test/functional/tests/cli/test_zero_metadata_command.py

```diff
@@ -0,0 +1,264 @@
+#
+# Copyright(c) 2021 Intel Corporation
+# SPDX-License-Identifier: BSD-3-Clause-Clear
+#
+import time
+from datetime import timedelta
+
+import pytest
+
+from api.cas import casadm, cli_messages, cli
+from api.cas.cache_config import CacheMode, CleaningPolicy
+from core.test_run import TestRun
+from storage_devices.disk import DiskTypeSet, DiskType, DiskTypeLowerThan
+from test_tools.disk_utils import get_device_filesystem_type, Filesystem
+from test_tools.fio.fio import Fio
+from test_tools.fio.fio_param import IoEngine, ReadWrite
+from test_utils.disk_finder import get_system_disks
+from test_utils.output import CmdException
+from test_utils.size import Size, Unit
+
+
+@pytest.mark.require_disk("cache", DiskTypeSet([DiskType.optane, DiskType.nand]))
+@pytest.mark.require_disk("core", DiskTypeLowerThan("cache"))
+def test_zero_metadata_negative_cases():
+    """
+        title: Test for '--zero-metadata' negative cases.
+        description: |
+          Test for '--zero-metadata' scenarios with expected failures.
+        pass_criteria:
+          - Zero metadata without '-force' failed when run on cache.
+          - Zero metadata failed when run on system drive.
+          - After zero metadata (success) on stopped cache load cache failed.
+    """
+    with TestRun.step("Prepare cache and core devices."):
+        cache_dev, core_dev = prepare_devices()
+
+    with TestRun.step("Start cache."):
+        cache = casadm.start_cache(cache_dev, CacheMode.WT, force=True)
+
+    with TestRun.step("Try to zero metadata and validate error message."):
+        output = TestRun.executor.run(cli.zero_metadata_cmd(str(cache_dev.path)))
+        if output.exit_code == 0:
+            TestRun.LOGGER.error("Zeroing metadata should fail!")
+        else:
+            cli_messages.check_stderr_msg(output, cli_messages.unavailable_device)
+            TestRun.LOGGER.info("Cannot zero metadata as expected.")
+
+    with TestRun.step("Try to zero metadata on system disk."):
+        os_disks = get_system_disks()
+        for os_disk in os_disks:
+            output = TestRun.executor.run(cli.zero_metadata_cmd(str(os_disk)))
+            if output.exit_code == 0:
+                TestRun.LOGGER.error("Zeroing metadata should fail!")
+            else:
+                cli_messages.check_stderr_msg(output, cli_messages.error_handling)
+                TestRun.LOGGER.info("Cannot zero metadata on system disk as expected.")
```

Comment:
This conditional statement doesn't look fine to me. The default message shall be error message and I think You should place here try-except statement instead.

Reply:
Good point to changing order of messages (else for error), thanks!
But for the try-except I think it is not suitable here (or I should perform commands different for that - is that what you meant? Using "TestRun.executor.run()" ends well (we are getting output from the command which were perform and not the 'CmdException' for catch) and the output consists error message - which we want to check.

Reply:
I think I meant something like this:
``` python
output = TestRun.executor.run(cli.zero_metadata_cmd(str(os_disk)))
try:
    cli_messages.check_stderr_msg(output, cli_messages.error_handling)
    TestRun.LOGGER.info("Cannot zero metadata on system disk as expected.")
except CmdException:    # not sure about the name of this custom exception
    TestRun.LOGGER.error("Zeroing metadata should fail!")
```
because `cli_messages.check_stderr_msg` can raise error.

Reply:
I've checked this, still the same - for failing case command performed correctly with error information logged "ERROR: Message is incorrect,..." and is not catched as CmdException.

Reply:
Instead of using cli.zero_metadata_cmd() use casadm.zero_metadata() and try..except approach